modules/library/src/main/scala/zio/elasticsearch/query/Queries.scala

```diff
@@ -593,6 +596,82 @@ private[elasticsearch] final case class MatchPhrasePrefix[S](field: String, valu
     Obj("match_phrase_prefix" -> Obj(fieldPath.foldRight(field)(_ + "." + _) -> Obj("query" -> value.toJson)))
 }
 
+sealed trait MultiMatchQuery[S]
+    extends ElasticQuery[S]
+    with HasBoost[MultiMatchQuery[S]]
+    with HasMinimumShouldMatch[MultiMatchQuery[S]] {
+
+  /**
+   * Sets the `fields` parameter for this [[zio.elasticsearch.query.ElasticQuery]]. The `fields` parameter is the array
+   * of fields that will be searched.
+   *
+   * @param fields
+   *   a array of fields to set `fields` parameter to
+   * @return
+   *   a new instance of the [[zio.elasticsearch.query.ElasticQuery]] with the `fields` array set.
+   */
+  def fields(field: String, fields: String*): MultiMatchQuery[S]
+
+  /**
+   * Sets the type-safe `fields` parameter for this [[zio.elasticsearch.query.ElasticQuery]]. The `fields` parameter is
+   * the type-safe array of fields that will be searched.
+   *
+   * @param fields
+   *   a type-safe array of fields to set `fields` parameter to
+   * @return
+   *   a new instance of the [[zio.elasticsearch.query.ElasticQuery]] with the `fields` array set.
+   */
+  def fields[S1 <: S: Schema](field: Field[S1, String], fields: Field[S1, String]*): MultiMatchQuery[S1]
+
+  /**
+   * Sets the `type` parameter for this [[zio.elasticsearch.query.ElasticQuery]]. The `type` parameter decides the way
+   * [[zio.elasticsearch.query.ElasticQuery]] is executed internally.
+   *
+   * @param matchingType
+   *   the [[zio.elasticsearch.query.MultiMatchType]] value of 'type' parameter
```

Comment:
Can you explain possible values for `matchingType`.
You can lookup for other places with similar @param.